src/Raven.Server/Documents/ETL/Providers/OLAP/OlaptEtl.cs

```diff
@@ -0,0 +1,202 @@
+        }
+
+        protected override IEnumerator<ToOlapItem> ConvertTimeSeriesEnumerator(DocumentsOperationContext context, IEnumerator<TimeSeriesSegmentEntry> timeSeries, string collection)
+        {
+            // RavenDB-16308
+            throw new NotSupportedException("Time Series are currently not supported by OLAP ETL");
+        }
+
+        protected override IEnumerator<ToOlapItem> ConvertTimeSeriesDeletedRangeEnumerator(DocumentsOperationContext context, IEnumerator<TimeSeriesDeletedRangeItem> timeSeries, string collection)
+        {
+            throw new NotSupportedException("Time series deletes aren't supported by OLAP ETL");
+        }
+
+        protected override void AfterAllBatchesCompleted(long batchTime)
+        {
+            var batchTimeMs = batchTime / 10_000;
+            UpdateEtlProcessState(LastProcessState, batchTimeMs);
+        }
+
+        public override void NotifyAboutWork(DatabaseChange change)
+        {
+            // intentionally not setting _waitForChanges here
+            // _waitForChanges is being set by the timer
+        }
+
+        protected override bool ShouldTrackAttachmentTombstones()
+        {
+            return false;
+        }
+
+        protected override bool ShouldFilterOutHiLoDocument()
+        {
+            return true;
+        }
+
+        public override bool ShouldTrackCounters() => false;
+
+        public override bool ShouldTrackTimeSeries() => false;
+
+        protected override EtlTransformer<ToOlapItem, OlapTransformedItems> GetTransformer(DocumentsOperationContext context)
+        {
+            return new OlapDocumentTransformer(Transformation, Database, context, Configuration);
+        }
+
+        protected override int LoadInternal(IEnumerable<OlapTransformedItems> records, DocumentsOperationContext context)
+        {
+            var count = 0;
+
+            foreach (var transformed in records)
+            {
+                var localPath = GetPath(transformed, out var remotePath);
+                count += transformed.GenerateFileFromItems(localPath, Logger);
+
+                using (Stream fileStream = File.OpenRead(localPath))
+                {
+                    UploadToDestination(fileStream, remotePath);
+                }
+
+                if (Configuration.KeepFilesOnDisk)
+                    continue;
+                
+                File.Delete(localPath);
+            }
+
+            return count;
+        }
+
+        private void UploadToDestination(Stream stream, string path)
+        {
+            if (_s3Settings != null)
+            {
+                var key = GetKey(_s3Settings, path);
+                UploadToS3(_s3Settings, stream, key);
+            }
+        }
+
+        private void UploadToS3(S3Settings s3Settings, Stream stream, string key)
+        {
+            using (var client = new RavenAwsS3Client(s3Settings, progress: null, Logger, CancellationToken))
```

Comment:
`progress: null` - can we expose it to the user? We already know how many files has been uploaded, and from here we might know the upload progress of current file, right?

If this is feasible please create a dedicates issue to handle that

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-16341